tools/brl_checks.c

```diff
@@ -60,155 +60,192 @@ print_widechars(widechar *buffer, int length) {
 	free(result_buf);
 }
 
-/* direction, 0=forward, otherwise backwards. If diagnostics is 1 then
+/* direction, 0=forward, 1=backwards, 2=both directions. If diagnostics is 1 then
  * print diagnostics in case where the translation is not as
  * expected */
+
 int
 check_base(const char *tableList, const char *input, const char *expected,
 		optional_test_params in) {
-	widechar *inbuf, *outbuf, *expectedbuf;
-	int inlen = strlen(input);
-	int outlen = inlen * 10;
-	int expectedlen = strlen(expected);
-	int i, retval = 0;
-	int funcStatus = 0;
-	formtype *typeformbuf = NULL;
-	int *inputPos = NULL;
-	int *outputPos = NULL;
-	int cursorPos = 0;
 
-	inbuf = malloc(sizeof(widechar) * inlen);
-	outbuf = malloc(sizeof(widechar) * outlen);
-	expectedbuf = malloc(sizeof(widechar) * expectedlen);
-	if (in.typeform != NULL) {
-		typeformbuf = malloc(outlen * sizeof(formtype));
-		memcpy(typeformbuf, in.typeform, outlen * sizeof(formtype));
-	}
-	if (in.cursorPos >= 0) {
-		cursorPos = in.cursorPos;
-	}
-	if (in.max_outlen >= 0) {
-		outlen = in.max_outlen;
-	}
-	inlen = _lou_extParseChars(input, inbuf);
-	if (!inlen) {
-		fprintf(stderr, "Cannot parse input string.\n");
-		retval = 1;
-		goto fail;
-	}
-	if (in.expected_inputPos) {
-		inputPos = malloc(sizeof(int) * outlen);
+	int i, retval = 0;
+	int direction = in.direction;
+	const int *expected_inputPos = in.expected_inputPos;
+	const int *expected_outputPos = in.expected_outputPos;
+	if (in.direction < 0 || in.direction > 2) {
+		fprintf(stderr, "Invalid direction.\n");
+		return 1;
 	}
-	if (in.expected_outputPos) {
-		outputPos = malloc(sizeof(int) * inlen);
+	if (in.direction != 0 && in.typeform != NULL) {
+		// Currently, in backward translation, nothing is done with the initial value of
+		// the
+		// typeform argument, and on returnal it always contains all zeros, so it doesn't
```

Comment:
"on return" (typo)

Reply:
It's not a typo. I actually thought this was a word (but apparently it's only the album by Oneohtrix Point Never).